### PR TITLE
Add documentation about custom samplers

### PR DESCRIPTION
**Pull request by iNikem**

### CODE REVIEW

File: docs/samplers.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
+## Sampling configuration
+
+| System property          | Environment variable     | Default value  | Support | Description |
+| ------------------------ | ------------------------ | -------------- | ------- | ----------- |
+| `otel.traces.sampler`    | `OTEL_TRACES_SAMPLER`     | `always_on`    | Stable  | The sampler to use for tracing.	|
+
+Splunk Distribution of OpenTelemetry Java supports all standard samplers as provided by
+[OpenTelemetry Java SDK Autoconfigure](https://github.com/open-telemetry/opentelemetry-java/tree/main/sdk-extensions/autoconfigure#sampler).
+In addition, we add the following two samplers.
```
Comment (theletterf) [inline docs/samplers.md:9]:
```suggestion
In addition, the distribution adds the following samplers:
```

File: docs/samplers.md [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+### internal_root_off
+Setting `otel.traces.sampler` to `internal_root_off` will drop all traces with root spans where `spanKind` is `INTERNAL`, `CLIENT` or `PRODUCER`.
+This setting only keeps root spans where `spanKind` is `SERVER` and `CONSUMER`.
```
Comment (theletterf) [inline docs/samplers.md:13]:
```suggestion
Setting `otel.traces.sampler` to `internal_root_off` drops all traces with root spans where `spanKind` is `INTERNAL`, `CLIENT` or `PRODUCER`. This setting only keeps root spans where `spanKind` is `SERVER` and `CONSUMER`.
```

File: docs/samplers.md [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+### rules
+This sampler allows to ignore individual endpoints and drop all traces that would originate from them.
+It applies only to spans with `SERVER` kind.
+For example, the following configuration results in all requests to `/healthcheck` to be excluded from monitoring:
```
Comment (theletterf) [inline docs/samplers.md:18]:
```suggestion
This sampler allows to ignore individual endpoints and drop all traces that originate from them.
It applies only to spans with `SERVER` kind.

For example, the following configuration results in all requests to `/healthcheck` to be excluded from monitoring:
```

**Review by theletterf — APPROVED** [review]

**Review by mateuszrzeszutek — APPROVED** [review]

**Comment by mateuszrzeszutek** [comment]
@iNikem can you rebase with the main branch? I've just merged a fix to all those smoke test failures

**Comment by iNikem** [comment]
> @iNikem can you rebase with the main branch? I've just merged a fix to all those smoke test failures

done

**Pull request merged**